ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/KSqlCorsHandler.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.CorsHandler;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+public class KSqlCorsHandler implements Handler<RoutingContext> {
+
+  private static final List<String> DEFAULT_ALLOWED_METHODS = Arrays.asList("GET", "POST", "HEAD");
+  private static final List<String> DEFAULT_ALLOWED_HEADERS = Arrays
+      .asList("X-Requested-With", "Content-Type", "Accept", "Origin");
+  private static final List<String> EXCLUDED_PATH_PREFIXES = Collections.singletonList("/ws/");
+
+  static void setupCorsHandler(final Server server, final Router router) {
+    final ApiServerConfig apiServerConfig = server.getConfig();
+    final String allowedOrigins = apiServerConfig
+        .getString(ApiServerConfig.CORS_ALLOWED_ORIGINS);
+    if (allowedOrigins.trim().isEmpty()) {
+      return;
+    }
+    final String convertedPattern = convertAllowedOrigin(allowedOrigins);
+    final CorsHandler corsHandler = CorsHandler.create(convertedPattern);
+    final List<String> allowedMethods = apiServerConfig
+        .getList(ApiServerConfig.CORS_ALLOWED_METHODS);
+    final Set<String> allowedMethodsSet = new LinkedHashSet<>(allowedMethods);
+    allowedMethodsSet.addAll(DEFAULT_ALLOWED_METHODS);
```

Comment:
These defaults are coming from the [Jetty defaults](https://www.eclipse.org/jetty/documentation/current/cross-origin-filter.html#cross-origin-config), right? Are they really additive like this? My interpretation of the docs was that these default values are used if the methods or headers are not explicitly supplied, but if something explicit is supplied then the defaults are ignored.

Otherwise the default of `*` for the allowed origins doesn't make sense.

Reply:
Ack